### PR TITLE
Make sure we are always throwing an exception from Core

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: tests/Unit/ExceptionTest.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class ExceptionTest extends TestCase
+{
+    public function testBranchAlias()
+    {
+        $finder = new Finder();
+        $root = \dirname(__DIR__, 2);
+        $finder
+            ->files()
+            ->in($root . '/src/Core/src')
+            ->in($root . '/src/Service/*/src')
```
Comment (Nyholm) [inline tests/Unit/ExceptionTest.php:23]: I dont check integrations

File: src/Core/src/Exception/UnexpectedValue.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Core\Exception;
+
+class UnexpectedValue extends \UnexpectedValueException implements Exception
```
Comment (jderusse) [inline src/Core/src/Exception/UnexpectedValue.php:7]: `UnexpectedValueException`

Comment (Nyholm) [reply src/Core/src/Exception/UnexpectedValue.php:7]: Are you sure? We usually dont use "Exception" suffix. 

Screenshot 2020-11-03 at 11 23 46

Comment (jderusse) [reply src/Core/src/Exception/UnexpectedValue.php:7]: ohh.. I was trapped by `RuntimeException` and `LogicException`... We are not consistent :cry:

Comment (Nyholm) [reply src/Core/src/Exception/UnexpectedValue.php:7]: True. But what we currently have sounds best. 

This is not super nice: 
```php
throw new Logic('Wrong');
```

**Comment by Nyholm** [comment]
Let's not merge this until Symfony 5.2 is released.

**Review by jderusse — APPROVED** [review]

**Comment by Nyholm** [comment]
Oh, no need to wait. 

The only new thing is `UnexpectedValue`, and no other packages are using that.

**Pull request merged**